GoDCR crashes if TAB used to move between fields on "name wallet" screen after seed words input

If you restore a wallet using seed words, the screen after the 33 seed words are entered cannot handle a "TAB" on keyboard.

Error output:

``` % ./godcr
Config file (/Users/testuser/Library/Application Support/Godcr/godcr.conf) does not exist. Using defaults.
2021-12-28 07:43:06.911 [INF] DLWL: Loaded 0 wallets
panic: runtime error: index out of range [-1]

goroutine 50 [running]:
github.com/planetdecred/godcr/ui/page/wallets.(*Restore).Handle(0xc0002ac400)
        /Users/testuser/godcr/ui/page/wallets/restore_page.go:480 +0x8d4
github.com/planetdecred/godcr/ui.(*Window).layoutPage.func2({{{0x0, 0x0}, {0x640, 0x4b0}}, {0x40000000, 0x40000000}, {0x56d4d60, 0xc000881298}, {0xc06aea1826c15f18, 0x1400c0801f, ...}, ...})
        /Users/testuser/godcr/ui/window.go:237 +0x2c
gioui.org/layout.Stack.Layout({0xb0}, {{{0x640, 0x4b0}, {0x640, 0x4b0}}, {0x40000000, 0x40000000}, {0x56d4d60, 0xc000881298}, {0xc06aea1826c15f18, ...}, ...}, ...)
        /Users/testuser/go/pkg/mod/gioui.org@v0.0.0-20211011183043-05f0f5c20f45/layout/stack.go:60 +0x895
github.com/planetdecred/godcr/ui.(*Window).layoutPage(0xc000254a80, {{{0x640, 0x4b0}, {0x640, 0x4b0}}, {0x40000000, 0x40000000}, {0x56d4d60, 0xc000881298}, {0xc06aea1826c15f18, ...}, ...}, ...)
        /Users/testuser/godcr/ui/window.go:232 +0x399
github.com/planetdecred/godcr/ui.(*Window).Loop(0xc000254a80, 0xc000881200, 0x0)
        /Users/testuser/godcr/ui/window.go:361 +0x3a9
created by main.main
        /Users/testuser/godcr/main.go:73 +0x4c5
```

--- a/ui/modal/create_password_modal.go
+++ b/ui/modal/create_password_modal.go
@@ -222,6 +222,9 @@ func (cm *CreatePasswordModal) Handle() {
 	cm.btnNegative.SetEnabled(!cm.isLoading)
 	if cm.btnNegative.Clicked() {
 		if !cm.isLoading {
+			if cm.parent != nil {
+				cm.parent.OnResume()
+			}
 			cm.Dismiss()
 		}
 	}

--- a/ui/modal/create_password_modal.go
+++ b/ui/modal/create_password_modal.go
@@ -76,6 +76,8 @@ func NewCreatePasswordModal(l *load.Load) *CreatePasswordModal {
 	th := material.NewTheme(gofont.Collection())
 	cm.materialLoader = material.Loader(th)
 
+	cm.Load.EnableKeyEventOnInfoModal = true
+
 	return cm
 }
 
@@ -89,11 +91,10 @@ func (cm *CreatePasswordModal) OnResume() {
 	} else {
 		cm.passwordEditor.Editor.Focus()
 	}
-	cm.Load.EnableKeyEvent = true
 }
 
 func (cm *CreatePasswordModal) OnDismiss() {
-	cm.Load.EnableKeyEvent = false
+	cm.Load.EnableKeyEventOnInfoModal = false
 }
 
 func (cm *CreatePasswordModal) Show() {
@@ -229,7 +230,6 @@ func (cm *CreatePasswordModal) Handle() {
 	} else {
 		decredmaterial.SwitchEditors(cm.keyEvent, cm.passwordEditor.Editor, cm.confirmPasswordEditor.Editor)
 	}
-
 }
 
 func (cm *CreatePasswordModal) passwordsMatch(editors ...*widget.Editor) bool {

--- a/ui/modal/create_password_modal.go
+++ b/ui/modal/create_password_modal.go
@@ -39,6 +39,8 @@ type CreatePasswordModal struct {
 	serverError string
 	description string
 
+	parent load.Page
+
 	materialLoader material.LoaderStyle
 
 	btnPositve  decredmaterial.Button
@@ -161,6 +163,12 @@ func (cm *CreatePasswordModal) validToCreate() bool {
 
 	return nameValid && editorsNotEmpty(cm.passwordEditor.Editor, cm.confirmPasswordEditor.Editor) &&
 		cm.passwordsMatch(cm.passwordEditor.Editor, cm.confirmPasswordEditor.Editor)
+}
+
+// SetParent sets the page that created PasswordModal as it's parent.
+func (cm *CreatePasswordModal) SetParent(parent load.Page) *CreatePasswordModal {
+	cm.parent = parent
+	return cm
 }
 
 func (cm *CreatePasswordModal) Handle() {

--- a/ui/page/wallets/restore_page.go
+++ b/ui/page/wallets/restore_page.go
@@ -111,6 +111,9 @@ func (pg *Restore) ID() string {
 }
 
 func (pg *Restore) OnResume() {
+	pg.Load.EnableKeyEvent = true
+	pg.keyEvent = pg.Receiver.KeyEvents
+
 }
 
 func (pg *Restore) Layout(gtx layout.Context) layout.Dimensions {
@@ -436,10 +439,14 @@ func (pg *Restore) Handle() {
 			return
 		}
 
+		pg.Load.EnableKeyEvent = false
+		pg.keyEvent = nil
+
 		modal.NewCreatePasswordModal(pg.Load).
 			Title("Enter wallet details").
 			EnableName(true).
 			ShowWalletInfoTip(true).
+			SetParent(pg).
 			PasswordCreated(func(walletName, password string, m *modal.CreatePasswordModal) bool {
 				go func() {
 					_, err := pg.WL.MultiWallet.RestoreWallet(walletName, pg.seedPhrase, password, dcrlibwallet.PassphraseTypePass)

